modules/client/src/main/java/org/apache/ignite/internal/client/ClientClusterNode.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.client;
+
+import org.apache.ignite.internal.tostring.S;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.NetworkAddress;
+import org.apache.ignite.network.NodeMetadata;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Client representation of a node in a cluster.
+ */
+public class ClientClusterNode implements ClusterNode {
+    /** Local ID assigned to the node instance. The ID changes between restarts. */
+    private final String id;
+
+    /** Unique name of a cluster member. Consistent between restarts. */
+    private final String name;
+
+    /** Network address of the node. */
+    private final NetworkAddress address;
+
+    /** Metadata of this node. */
+    @Nullable
+    private final NodeMetadata nodeMetadata;
+
+    /**
+     * Constructor.
+     *
+     * @param id      Local id that changes between restarts.
+     * @param name    Unique name of a member in a cluster.
+     * @param address Node address.
+     * @param nodeMetadata Node metadata.
+     */
+    public ClientClusterNode(String id, String name, NetworkAddress address, @Nullable NodeMetadata nodeMetadata) {
+        this.id = id;
+        this.name = name;
+        this.address = address;
+        this.nodeMetadata = nodeMetadata;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param id      Local ID that changes between restarts.
+     * @param name    Unique name of a cluster member.
+     * @param address Node address.
+     */
+    public ClientClusterNode(String id, String name, NetworkAddress address) {
+        this(id, name, address, null);
+    }
+
+    /** {@inheritDoc} */
```

Comment:
By the way, we don't have to have these comments

Reply:
Yep, you are right. In accordance with https://google.github.io/styleguide/javaguide.html#s7.3.2-javadoc-exception-overrides there is no special need to explicitly mark a method with `inheritDoc` tag. Will remove redundant javadocs.